ecdsa/src/recovery.rs

```diff
@@ -285,7 +285,18 @@ where
     ) -> Result<Self> {
         let (r, s) = signature.split_scalars();
         let z = <Scalar<C> as Reduce<C::Uint>>::reduce_bytes(&bits2field::<C>(prehash)?);
-        let R = AffinePoint::<C>::decompress(&r.to_repr(), u8::from(recovery_id.is_y_odd()).into());
+
+        let mut r_bytes = r.to_repr();
+        if recovery_id.is_x_reduced() {
+            r_bytes = match Option::<C::Uint>::from(
```

Comment:
Is there some reason you can't assign `r_bytes` in the `Some` match arm? It looks like that's the only case that matters.

I guess it couldn't infer `Option` with `into()`? If not, I would think `Option::from` would still work without the explicit `C::Uint` annotation for the generic type parameter.

Reply:
It is possible to assign inside the some/match arm. I'll fix that.

I thought it could infer the type too, but the compiler complains that it can't infer the type of `restored` without the explicit generic type parameter.